Sources/Utility/Process.swift

```diff
@@ -165,6 +171,21 @@ public final class Process: ObjectIdentifierProtocol {
         self.redirectOutput = redirectOutput 
     }
 
+    /// Validate that the program provided exists on the search path and is executable.
+    func validateArgumentIsExecutable(_ program: String) throws {
+
+        // Check to see if the argument is already listed in the vaildated cache.
+        guard !Process.validatedExecutables.contains(program) else {
+            return
+        }
+        let envSearchPaths = Utility.getEnvSearchPaths(pathString: getenv("PATH"), currentWorkingDirectory: currentWorkingDirectory)
```

Comment:
actually getenv and currentworkingdir is also not thread safe, i'll comment here again on how to make this thread safe.